spark/src/main/java/org/apache/iceberg/spark/source/BaseDataReader.java

```diff
@@ -125,7 +125,7 @@ public void close() throws IOException {
 
   protected InputFile getInputFile(FileScanTask task) {
     Preconditions.checkArgument(!task.isDataTask(), "Invalid task type");
-    return inputFiles.get(task.file().path().toString());
+    return getInputFile(task.file().path().toString());
```

Comment:
I don't think this needs to change. Can you go back to the previous implementation?

Reply:
Done.